src/dmd/backend/cv8.d

```diff
@@ -5,65 +5,82 @@
  * Copyright:    Copyright (C) 2012-2018 by The D Language Foundation, All Rights Reserved
  * Authors:     $(LINK2 http://www.digitalmars.com, Walter Bright)
  * License:     $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
- * Source:      $(LINK2 https://github.com/dlang/dmd/blob/master/src/dmd/backend/code.c, backend/code.c)
+ * Source:      $(LINK2 https://github.com/dlang/dmd/blob/master/src/dmd/backend/cv8.d, backend/cv8.d)
  */
 
 // This module generates the .debug$S and .debug$T sections for Win64,
 // which are the MS-Coff symbolic debug info and type debug info sections.
 
-#if !SPP
-
-#include        <stdio.h>
-#include        <string.h>
-#include        <stdlib.h>
-#include        <time.h>
-
-#include        "cc.h"
-#include        "el.h"
-#include        "code.h"
-#include        "oper.h"
-#include        "global.h"
-#include        "type.h"
-#include        "dt.h"
-#include        "exh.h"
-#include        "cgcv.h"
-#include        "cv4.h"
-#include        "obj.h"
-#include        "outbuf.h"
-#include        "varstats.h"
-
-static char __file__[] = __FILE__;      /* for tassert.h                */
-#include        "tassert.h"
-
-#if _MSC_VER || __sun
-#include        <alloca.h>
-#endif
-
-#define null NULL
-typedef unsigned char ubyte;
-typedef unsigned short ushort;
-#define mask(m) (1 << (m))
-
-#if MARS
-#if TARGET_WINDOS
+module dmd.backend.cv8;
+
+version (MARS)
+{
+
+import core.stdc.stdio;
+import core.stdc.stdlib;
+import core.stdc.string;
+extern (C) char* getcwd(char*, size_t);
+
+import dmd.backend.cc;
+import dmd.backend.cdef;
+import dmd.backend.cgcv;
+import dmd.backend.code;
+import dmd.backend.code_x86;
+import dmd.backend.cv4;
+import dmd.backend.memh;
+import dmd.backend.el;
+import dmd.backend.exh;
+import dmd.backend.global;
+import dmd.backend.obj;
+import dmd.backend.oper;
+import dmd.backend.outbuf;
+import dmd.backend.rtlsym;
+import dmd.backend.ty;
+import dmd.backend.type;
+import dmd.backend.varstats;
+import dmd.backend.xmm;
+
+extern (C++):
+
+static if (TARGET_WINDOS)
+{
+
+int REGSIZE();
+extern (D) uint mask(uint m); //{ return 1 << m; }
+
+// Determine if this Symbol is stored in a COMDAT
+bool symbol_iscomdat(Symbol* s)
+{
+    version (MARS)
```

Comment:
Isn't this redundant given this is already in `version (MARS)` ?

Reply:
Yes, but there are problems with where it should go. Needs a refactoring, but first translate, then fix/refactor.